pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -54,7 +54,11 @@ patchShebangs() {
                 echo "$f: interpreter directive changed from \"$oldInterpreterLine\" to \"$newInterpreterLine\""
                 # escape the escape chars so that sed doesn't interpret them
                 escapedInterpreterLine=$(echo "$newInterpreterLine" | sed 's|\\|\\\\|g')
+                # Preserve times, see: https://github.com/NixOS/nixpkgs/pull/33084
```

Comment:
Golfing: `mtime=$(stat -c '%Y' "$f"); whatever; touch -d @$mtime "$f"` (I think that does the same thing, works with GNU, at least ...)

Reply:
It won't, it doesn't preserve nanoseconds.

Reply:
That's a good idea Just to `local mtime=...` so that it doesn't pollute the global scope.

Reply:
`stat -c '%y'` then?

Reply:
It seems error-prone to convert `%y` format into format accepted by `touch`. If pollution is a problem, `mktemp` could be used to reserve a unique file name in `/tmp` and then that one file can be used as a buffer.

Reply:
On my end, `touch -d "$(stat -c '%y' ...)" ...` works as I expect, in that the mtimes are identical, to the nanosecond, without the need for conversion. Same for `touch -d @$(stat -c '%.Y')`. Perhaps these are GNU-isms, not to be relied on?

Reply:
Do we have these options in the bootstrapping phase (in case this setup-hook is used there)?

Reply:
```
touch -d @$(stat -c '%Y' README.md)  default.nix
```

in the nixpkgs repo also works with busybox. There is no posix version of the `stat` command so there is no standard set of flags we would have to rely on.

Reply:
We're not restricted to POSIX, since stdenv supplies coreutils.